deviceauth.go

```diff
@@ -112,6 +112,7 @@ func retrieveDeviceAuth(ctx context.Context, c *Config, v url.Values) (*DeviceAu
 	}
 
 	body, err := io.ReadAll(io.LimitReader(r.Body, 1<<20))
+	r.Body.Close()
 	if err != nil {
```

Comment:
it should be `defer r.Body.Close()` before reading all the content.

```
defer r.Body.Close()
body, err := io.ReadAll(io.LimitReader(r.Body, 1<<20))
```

Reply:
Hm, I'm not sure why we need to use `defer`, unless `io.ReadAll(io.LimitReader(r.Body, 1<<20))` can panic?  Anyway, I implemented your suggestion in https://github.com/golang/oauth2/compare/ffeca96aa76167a098068334a567bb7c0975acc8..8499b419ad4097ac16cd5b264b8efe53791ae259.  Thanks for reviewing!

Reply:
Your solution is not wrong, but it's very common in golang to use the `defer reader.Close()` pattern.
A more precise version that can handle all kind of errors would be:

```
body, err := io.ReadAll(io.LimitReader(r.Body, 1<<20))
if err != nil {
  r.Body.Close() // ignore _this_ error - we already have one
  ...
  return ...
}
err = r.Body.Close()
if err != nil {
  return ...
}
```

But this would make the code more unreadable. In most cases it is _ok_ to ignore the close error.